components/eventing-controller/pkg/ems2/auth/config.go

```diff
@@ -0,0 +1,15 @@
+package auth
+
+import (
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/env"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+// Config returns a new oauth2 client credentials config instance.
+func getDefaultOauth2Config(cfg *env.Config) *clientcredentials.Config {
```

Comment:
```suggestion
func getDefaultOauth2Config(cfg env.Config) *clientcredentials.Config {
```

either pass by value, or check for nil

Reply:
In our case it cannot be null, see here:
```
func GetConfig() *Config {
	cfg := new(Config)
	if err := envconfig.Process("", cfg); err != nil {
		log.Fatalf("Invalid configuration: %v", err)
	}
	return cfg
}
```